### ISSUE TITLE
sonos fails on parsing DeviceProperties1.xml

### ISSUE THREAD

**Issue by RanTalbott** [avantrec/soco-cli#61]
Ubuntu 20.04
python 3.8.10
Just installed soco-0.29.1 and soco-cli-0.4.70 via pip
Sonos Play:1 from a thrift store. Works okay with noson, but I dislike the app. I did some experimenting with noson's cli application and got info on the Play:1's version:
(SONOS)Request: SoftwareVersion = 74.0-43312
(SONOS)Request: DisplaySoftwareVersion = 15.7
(SONOS)Request: HardwareVersion = 1.20.1.6-1.2
(SONOS)Request: IPAddress = 192.168.0.23
(SONOS)Request: MACAddress = 94:9F:3E:16:0A:E0
(SONOS)Request: CopyrightInfo = © 2003-2023, Sonos, Inc. All rights reserved.

I set the SPKR variable, and I get the same results from 'sonos status'  'sonos "Living Roon" status' and 'sonos play_file filename':
`Error: Failed to parse: http://192.168.0.23:1400/xml/DeviceProperties1.xml`

So it found the Sonos, and appears to have downloaded the properties file, but doesn't like what it sees. I grabbed the file with wget, xmllint doesn't complain about the syntax, kwrite appears to display it correctly, and xmltodict.parse() makes a dict without complaint.

I did a search for the "Failed to parse" message in the soco and soco-cli source, but couldn't find it. So I can't tell who's unhappy, or why. I'm guessing that the version of firmware in my Sonos is producing an XML file with different tag names, or in a different order, or somesuch, but that's just a guess. I've attached a copy so someone can make a more informed analysis.

[DeviceProperties1.txt](https://github.com/avantrec/soco-cli/files/12812383/DeviceProperties1.txt)

**Comment by pwt** [comment]
Hi:

Have you set up the speaker with one of the Sonos apps? If so, does the speaker work OK with that app?

In general, SoCo requires that the Sonos speaker/system is properly configured in order to work correctly.

**Comment by RanTalbott** [comment]
Yes, I had it working fine with the noson app. But its GUI didn't handle my large CD collection well, and the CLI app is only partial and not scriptable, so I started looking for alternatives.

I did manage to find the cause of the problem. I decided to try the web app that's distributed with SoCo, to see if I could isolate the problem to the driver or the app. It gave me a complete traceback when it hit the failure, and it turned out to be in urllib3. It was actually the URL it was having trouble parsing, not the file it pointed to.

I have an old version of urllib3 installed: 1.25.8. When I installed soco and soco-cli in a virtual environment, pip pulled in 2.0.6. So the requirements need to be adjust to insist on a newer version that doesn't have that bug.
